cove/input/views.py

```diff
@@ -82,6 +82,10 @@ def data_input(request, form_classes=default_form_classes, text_file_name='test.
                     })
             elif form_name == 'text_form':
                 data.original_file.save(text_file_name, ContentFile(form['paste'].value()))
-            return redirect(data.get_absolute_url())
+            
+            if 'raw' in request_data:
```

Comment:
How does ```'raw'``` make it to ```request_data```? Are you sending a query string in the URL to the GET request when asking for the json response? If so, how? Manually, adding it to the URL for the existing data URL?

Also, note that you are not adding parameters to the upload forms, which means it won't be a ```'raw'``` in any of the POST requests.